script/tool/lib/src/build_examples_command.dart

```diff
@@ -225,6 +244,58 @@ class BuildExamplesCommand extends PackageLoopingCommand {
       }
     }
 
+    final File pluginToolsConfig =
+        example.directory.childFile(_pluginToolsConfigFileName);
+    if (pluginToolsConfig.existsSync()) {
+      final Object? configuration =
+          loadYaml(pluginToolsConfig.readAsStringSync());
+      if (configuration is! YamlMap) {
+        printError('The $_pluginToolsConfigFileName file must be a YAML map.');
+        printError(
+            'Currently, the key "$_pluginToolsConfigBuildFlagsKey" is the only one that has an effect.');
+        printError(
+            'It must itself be a map. Currently, in that map only the key "$_pluginToolsConfigGlobalKey"');
+        printError(
+            'has any effect; it must contain a list of arguments to pass to the');
+        printError('flutter tool.');
+        printError(_pluginToolsConfigExample);
+        throw ToolExit(_exitInvalidPluginToolsConfig);
+      }
+      if (configuration.containsKey(_pluginToolsConfigBuildFlagsKey)) {
+        final Object? buildFlagsConfiguration =
+            configuration[_pluginToolsConfigBuildFlagsKey];
+        if (buildFlagsConfiguration is! YamlMap) {
+          printError(
+              'The $_pluginToolsConfigFileName file\'s "$_pluginToolsConfigBuildFlagsKey" key must be a map.');
+          printError(
+              'Currently, in that map only the key "$_pluginToolsConfigGlobalKey" has any effect; it must ');
+          printError(
+              'contain a list of arguments to pass to the flutter tool.');
+          printError(_pluginToolsConfigExample);
+          throw ToolExit(_exitInvalidPluginToolsConfig);
+        }
+        if (buildFlagsConfiguration.containsKey(_pluginToolsConfigGlobalKey)) {
+          final Object? globalBuildFlagsConfiguration =
+              buildFlagsConfiguration[_pluginToolsConfigGlobalKey];
+          if (globalBuildFlagsConfiguration is! YamlList) {
+            printError(
+                'The $_pluginToolsConfigFileName file\'s "$_pluginToolsConfigBuildFlagsKey" key must be a map');
+            printError('whose "$_pluginToolsConfigGlobalKey" key is a list.');
+            printError(
+                'That list must contain a list of arguments to pass to the flutter tool.');
+            printError(
+                'For example, the $_pluginToolsConfigFileName file could look like:');
+            printError(_pluginToolsConfigExample);
+            throw ToolExit(_exitInvalidPluginToolsConfig);
+          }
+          extraBuildFlags = <String>[
+            ...extraBuildFlags,
+            ...globalBuildFlagsConfiguration.cast<String>()
+          ];
+        }
+      }
+    }
```

Comment:
Optional nit: with all the error checking this is long enough that I would extract it to a `List<String> _getPackageSpecificBuildFlags(RepositoryPackage package)` directory.

Reply:
ok, extracted to function